### PR TITLE
Added tasks error handling

### LINKED ISSUE #37: Exception handling for celery tasks
Common or plugin specific handlers for exceptions in celery tasks.

They should set the status to ERROR and clean things up if necessary.

Related to #19

### PR DESCRIPTION
**Pull request by roll**
- fixes #37

### CODE REVIEW

**Comment by roll** [comment]
@amercader 
I haven't finished but you could check the idea.

File: goodtablesio/plugins/github/tasks.py [hunk 1]
```diff
@@ -43,8 +43,14 @@ def get_validation_conf(clone_url, job_id):
...
 @signals.task_postrun.connect(sender=validate)
 def post_task_handler(**kwargs):
+    # We need to import the DB connection at this point, as it has been
+    # initialized when the worker started
+    from goodtablesio.tasks import tasks_db
 
-    job = kwargs['retval']
+    # We can't use retval because on errors
+    # it will be an exception
+    job_id = kwargs['kwargs']['job_id']
```
Comment (amercader) [inline goodtablesio/plugins/github/tasks.py]: Then we should only connect to the database if the retval is an exception to save a db lookup on 99% (hopefully) of cases were there isn't an exception. Or better yet can we modify the exception adding it extra info (the `job` object) on `on_failure`?

File: goodtablesio/tasks.py [hunk 2]
```diff
@@ -74,4 +119,3 @@ def validate(validation_conf, job_id):
                             ['job_id'],
                             types={'report': JSONB, 'finished': DateTime},
                             ensure=True)
-    return job
```
Comment (amercader) [inline goodtablesio/tasks.py]: See comment above. I think that would save a lot of db lookups

File: goodtablesio/plugins/github/tasks.py (hunk 1 above)
Comment (roll) [inline goodtablesio/plugins/github/tasks.py]: Good idea. I'll update. Though I don't think db calls will be a bottleneck at any point of the project.

Comment (amercader) [inline goodtablesio/plugins/github/tasks.py]: we need to think big :)

File: goodtablesio/models.py [hunk 3]
```diff
@@ -20,3 +21,4 @@ def __init__(self, *args, **kwargs):
         self.created = kwargs.get('created')
         self.finished = kwargs.get('finished')
         self.report = kwargs.get('report')
+        self.error = kwargs.get('report')
```
Comment (amercader) [inline goodtablesio/models.py]: Typo

File: goodtablesio/helpers/create.py [hunk 4]
```diff
@@ -41,7 +41,7 @@ def create_job(validation_conf, job_id=None):
     insert_job_row(job_id)
 
     # Create celery task
-    tasks.validate.apply_async((validation_conf, job_id))
+    tasks.validate.s(validation_conf, job_id=job_id).apply_async()
```
Comment (amercader) [inline goodtablesio/helpers/create.py]: Why do we need to call it like this? It seems like celery always has an uglier way of doing things :)

Comment (roll) [inline goodtablesio/helpers/create.py]: Changed to `delay`.

File: goodtablesio/models.py (hunk 3 above)
Comment (roll) [inline goodtablesio/models.py]: fixed

**Comment by amercader** [comment]
looks good @roll

**Pull request merged**